flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSink.java

```diff
@@ -270,15 +284,38 @@ public DynamicTableSinkSpec getTableSinkSpec() {
                     || (type.is(LogicalTypeRoot.VARCHAR)
                             && (LogicalTypeChecks.getLength(type) < VarCharType.MAX_LENGTH))) {
                 charFieldsAndLengths.add(
-                        new ConstraintEnforcer.CharFieldInfo(
+                        new ConstraintEnforcer.FieldInfo(
                                 i, LogicalTypeChecks.getLength(type), isChar));
             } else if (isChar) { // Should pad
-                charFieldsAndLengths.add(new ConstraintEnforcer.CharFieldInfo(i, null, isChar));
+                charFieldsAndLengths.add(new ConstraintEnforcer.FieldInfo(i, null, isChar));
             }
         }
         return charFieldsAndLengths;
     }
 
+    /**
+     * Returns a List of {@link ConstraintEnforcer.FieldInfo}, each containing the info needed to
+     * determine whether a binary value needs trimming and/or padding.
+     */
+    private List<ConstraintEnforcer.FieldInfo> getBinaryrFieldInfo(RowType physicalType) {
```

Comment:
in theory we could merge the two methods and parameterize with the type roots and lengths?